torchqdynamics/mesolve/mesolve.py

```diff
@@ -153,11 +137,7 @@ def mesolve(
 
     # get saved tensors and restore correct batching
     rho_save, exp_save = solver.y_save, solver.exp_save
-    if rho0.ndim == 2:
-        rho_save = rho_save.squeeze(1)
-        exp_save = exp_save.squeeze(1)
-    if H.ndim == 2:
-        rho_save = rho_save.squeeze(0)
-        exp_save = exp_save.squeeze(0)
+    rho_save = formatter.unbatch(rho_save)
+    exp_save = formatter.unbatch(exp_save)
```

Comment:
Found a small bug (already present in the previous code, but it's a good PR to fix it). If no `exp_ops` are passed, then `exp_save` is None, and this raises an error. So we need to check if `exp_save` is not None somewhere.

Same thing for `sesolve`.